workflow_imaging/ingest.py

```diff
@@ -1,68 +1,70 @@
-    sess_folder_names = []
-    for subj_key in subject.Subject.fetch('KEY'):
-        subj_dir = data_dir / subj_key['subject']
-        if subj_dir.exists():
-            for tiff_filepath in subj_dir.glob('*/*.tif'):
-                sess_folder = tiff_filepath.parent
-                if sess_folder.name not in sess_folder_names:
-                    tiff_filepaths = [fp.as_posix() for fp in sess_folder.glob('*.tif')]
-                    try:  # attempt to read .tif as a scanimage file
-                        loaded_scan = scanreader.read_scan(tiff_filepaths)
-                    except Exception as e:
-                        print(f'ScanImage loading error: {tiff_filepaths}\n{str(e)}')
-                        loaded_scan = None
-
-                    if loaded_scan is not None:
-                        sess_folder_names.append(sess_folder.name)
-                        recording_time = get_scanimage_acq_time(loaded_scan)
-                        header = parse_scanimage_header(loaded_scan)
-                        session_key = {**subj_key, 'session_datetime': recording_time}
-                        scanner = header['SI_imagingSystem']
-                        if session_key not in Session.proj():
-                            print(f'Inserting session: {session_key} (from {sess_folder})')
-                            scanners.append({'scanner': scanner})
-                            sessions.append(session_key)
-                            scans.append({**session_key, 'scan_id': 0, 'scanner': scanner})
+    session_directories, processing_tasks = [], []
 
-    Equipment.insert(scanners, skip_duplicates=True)
-    Session.insert(sessions)
-    scan.Scan.insert(scans)
-    print(f'Inserted {len(sessions)} session(s)')
+    for session in sessions_dict:
+        sess_dir = pathlib.Path(session['session_dir'])
+        tiff_filepaths = [fp.as_posix() for fp in sess_dir.glob('*.tif')]
 
-    # populate ScanInfo
-    print('---- Populate ScanInfo ----')
-    scan.ScanInfo.populate(suppress_errors=True, display_progress=True)
+        try:  # attempt to read .tif as a scanimage file
+            loaded_scan = scanreader.read_scan(tiff_filepaths)
+        except Exception as e:
+            print(f'ScanImage loading error: {tiff_filepaths}\n{str(e)}')
+            loaded_scan = None
+
+        if loaded_scan is not None:
+            recording_time = get_scanimage_acq_time(loaded_scan)
+            header = parse_scanimage_header(loaded_scan)
+            session_key = {'subject': session['subject'], 'session_datetime': recording_time}
+            scanner = header['SI_imagingSystem'].strip('\'')
+            if session_key not in Session.proj():
+                scanners.append({'scanner': scanner})
+                sessions.append(session_key)
+                scans.append({**session_key, 'scan_id': 0, 'scanner': scanner})
+                session_directories.append({**session_key, 'session_dir': session['session_dir']})
```

Comment:
this is where we need to extract the relative path of the `session_dir` (relative to the `imaging_root_data_dir`) and only insert the relative path, you can use `.relative_to()` (from `pathlib` library)

Reply:
thanks